Add WCAG autocomplete attributes to FormConfig

jakerarr opened the pull request:
- Add autocomplete constants file
- Update FormField with get/set for autoComplete attribute
- Update twig files to add autocomplete attribute when applicable

Commit:
Update selectAjax.twig data attributes
- Add raw to twig output

Commit:
Merge remote-tracking branch 'KongHack/master' into master

Commit:
Add autocomplete functionality to form config

Commit:
Update composer.lock

Commit:
Fix autoComplete line in twig files

Commit:
Reset composer.lock - i don't trust it

GameCharmer reviewed (changes requested):
Skip this one, move onto the below changes

GameCharmer commented on src/Core/AutoCompleteConstants.php:
```diff
@@ -0,0 +1,118 @@
+<?php
+namespace GCWorld\FormConfig\Core;
```
Move this to Interfaces.  It's 5 folders down. :P

jakerarr replied:
Moved to interfaces!

GameCharmer commented on src/Forms/FormField.php:
```diff
@@ -227,6 +233,27 @@ public function getNameRaw()
+     */
+    public function setAutoComplete($component)
+    {
+        if (!isset(AutoCompleteConstants::COMPONENTS[$component])) {
```
If you're going to force exceptions, you need to make sure ALL possible options exist, like "on" and "off"

jakerarr replied:
Added ON/OFF options!

GameCharmer commented on twig/BS3/field_parents/select2HTML.twig:
```diff
@@ -1,5 +1,11 @@
+           name="{{ field.name }}"
+           id="{{ field.id }}"
+           class="{{ field.class }}"
+           {{ (field.getAutoComplete() = '' ? 'autocomplete="off"' : 'autocomplete="' ~ field.getAutoComplete() ~ '"') }}
```
For every one of these that you're changing, make sure you're changing the accompanying PHP file in src/Fields to ensure the method exists.

You may want to add your autoComplete method to a Trait (in the traits folder), then USE it in FormField and in any other fields that will have the autoComplete method

jakerarr replied:
Added changes to PHP files.  Made trait and applied it to FormField and all php files where the associated twig file was changed!

Commit:
- Fix twig issues
- Move AutoComplete methods to trait
- Move AutoCompleteConstants to Interfaces

Commit:
- Move AutoComplete methods to trait
- Move AutoCompleteConstants to Interfaces
- Update twig errors

Commit:
Fix spacing on twig operators
- Fix issue with twig operators
